js/base/errors.js

```diff
@@ -17,19 +18,34 @@ function subclass (BaseClass, classes, namespace = {}) {
 
             [className]: class extends BaseClass {
 
-                constructor (message) {
+                constructor (errorMessage, verbose = false, exchangeId = undefined, httpStatusCode = undefined, httpStatusText = undefined, url = undefined, httpMethod = undefined, responseHeaders = undefined, responseBody = undefined, responseJson = undefined) {
```

Comment:
Suppose we need to add `exchangeMethod`, `methodArguments`, `requestHeaders` and `requestBody`, what would you prefer:

```
constructor (
    errorMessage,
    verbose = false,
    exchangeId = undefined,
    exchangeMethod = undefined,
    exchangeMethodArguments = undefined,
    httpStatusCode = undefined,
    httpStatusText = undefined,
    url = undefined,
    httpMethod = undefined,
    requestHeaders = undefined,
    requestBody = undefined,
    responseHeaders = undefined,
    responseBody = undefined,
    responseJson = undefined
)
```

or

```
constructor (
    errorMessage,
    verbose = false,
    params = {
        exchangeId = undefined,
        exchangeMethod = undefined,
        exchangeMethodArguments = undefined,
        httpStatusCode = undefined,
        httpStatusText = undefined,
        url = undefined,
        httpMethod = undefined,
        requestHeaders = undefined,
        requestBody = undefined,
        responseHeaders = undefined,
        responseBody = undefined,
        responseJson = undefined
    }
)
```

? )